Failed to execute goal com.querydsl:querydsl-maven-plugin:4.0.9.BUILD-SNAPSHOT:export

I'm getting this error when I try to build my project with the version in master.

This is the full stack trace:

```
[ERROR] Failed to execute goal com.querydsl:querydsl-maven-plugin:4.0.9.BUILD-SNAPSHOT:export (default) on project xxx: Unable to parse configuration of mojo com.querydsl:querydsl-maven-plugin:4.0.9.BUILD-SNAPSHOT:export: Cannot find setter, adder nor field in com.querydsl.maven.MetadataExportMojo for 'maven.querydsl.skip' -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.querydsl:querydsl-maven-plugin:4.0.9.BUILD-SNAPSHOT:export (default) on project xxx: Unable to parse configuration of mojo com.querydsl:querydsl-maven-plugin:4.0.9.BUILD-SNAPSHOT:export: Cannot find setter, adder nor field in com.querydsl.maven.MetadataExportMojo for 'maven.querydsl.skip'
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:221)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginConfigurationException: Unable to parse configuration of mojo com.querydsl:querydsl-maven-plugin:4.0.9.BUILD-SNAPSHOT:export: Cannot find setter, adder nor field in com.querydsl.maven.MetadataExportMojo for 'maven.querydsl.skip'
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.populatePluginFields(DefaultMavenPluginManager.java:597)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo(DefaultMavenPluginManager.java:529)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:92)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	... 19 more
Caused by: org.codehaus.plexus.component.configurator.ComponentConfigurationException: Cannot find setter, adder nor field in com.querydsl.maven.MetadataExportMojo for 'maven.querydsl.skip'
	at org.codehaus.plexus.component.configurator.converters.ComponentValueSetter.<init>(ComponentValueSetter.java:95)
	at org.codehaus.plexus.component.configurator.converters.composite.ObjectWithFieldsConverter.processConfiguration(ObjectWithFieldsConverter.java:158)
	at org.codehaus.plexus.component.configurator.BasicComponentConfigurator.configureComponent(BasicComponentConfigurator.java:56)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.populatePluginFields(DefaultMavenPluginManager.java:567)
	... 22 more
```

The only change I made was update the version number for Querydsl. Both for the maven plugin and the library.

I get a different error:
`[ERROR] Failed to execute goal com.querydsl:querydsl-maven-plugin:4.1.3.BUILD-SNAPSHOT:export (default) on project test: Unable to parse configuration of mojo com.querydsl:querydsl-maven-plugin:4.1.3.BUILD-SNAPSHOT:export: Cannot find setter, adder nor field in com.querydsl.maven.MetadataExportMojo for 'maven.querydsl.skip' -> [Help 1]
`
like the original poster.

Hmm, FWIW, I can reproduce it on a Linux box.
```
Apache Maven 3.0.5
Maven home: /usr/share/maven
Java version: 1.8.0_45-internal, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.19.0-22-generic", arch: "amd64", family: "unix"

...

[ERROR] Failed to execute goal com.querydsl:querydsl-maven-plugin:4.1.3.BUILD-SNAPSHOT:export (default) on project test: Unable to parse configuration of mojo com.querydsl:querydsl-maven-plugin:4.1.3.BUILD-SNAPSHOT:export: Cannot find setter, adder nor field in com.querydsl.maven.MetadataExportMojo for 'maven.querydsl.skip' -> [Help 1]

```

I found that Maven by default includes the (ancient) 2.9 version of `maven-plugin-plugin`
http://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugin-plugin/

In the meantime, there were some more releases, which fixed a small issue where it would take the `property=` name of the `@parameter` verbatim as property name.

@hylkevds Could you maybe verify if this fixes the issue? For me it does.

```
Apache Maven 3.0.5
Maven home: /usr/share/maven
Java version: 1.8.0_45-internal, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.19.0-22-generic", arch: "amd64", family: "unix"

...

[ERROR] Failed to execute goal com.querydsl:querydsl-maven-plugin:4.1.3.BUILD-SNAPSHOT:export (default) on project test: The parameters 'targetFolder', 'packageName' for goal com.querydsl:querydsl-maven-plugin:4.1.3.BUILD-SNAPSHOT:export are missing or invalid -> [Help 1]

```

--- a/querydsl-maven-plugin/pom.xml
+++ b/querydsl-maven-plugin/pom.xml
@@ -28,6 +28,18 @@
         <maven.javadoc.skip>true</maven.javadoc.skip>
     </properties>
 
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-plugin-plugin</artifactId>
+                    <version>3.4</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
     <dependencies>
         <dependency>
             <groupId>org.apache.maven</groupId>

